dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -423,6 +423,22 @@ instance Arbitrary Var where
 
     shrink = genericShrink
 
+embedComposesWithBinaryEncode :: forall a. (ToDhall a, FromDhall a, Eq a, Typeable a, Arbitrary a, Show a)
```

Comment:
Isn't this property implied when binary encoding and decoding, and embedding and extracting to/from Dhall both roundtrip?

I might be misreading it though. It makes my head spin! ;)